Add mcxa156 i3c support

NeilChen93 opened the pull request:
Add mcxa156 i3c support

NeilChen93 committed:
drivers: syscon: support mcxa156 i3c clock in syscon driver

Add mcxa156 i3c clock support

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>

NeilChen93 committed:
dts: arm/nxp: Add i3c nodes to NXP MCXA156 dtsi file

Add i3c nodes to NXP MCXA156 dtsi file

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>

kartben commented on samples/sensor/thermometer/sample.yaml:
```diff
@@ -10,6 +10,7 @@ tests:
       - robokit1              # ntc_thermistor
       - adi_eval_adin1110ebz  # adt7420
       - frdm_mcxn947/mcxn947/cpu0 # p3t1755
+      - frdm_mcxa156          # p3t1755
```
Drop. There is no need to add to CI load as this will not be testing anything that isn't already tested elsewhere (i3c driver tests, sensor driver test, or even the line just above--even if it should likely also go away)
https://docs.zephyrproject.org/latest/samples/sample_definition_and_criteria.html#twister-should-be-able-to-build-every-sample

kartben commented on samples/sensor/thermometer/boards/frdm_mcxa156.overlay:
This shouldn't be defined in an overlay if it's an on-board sensor. Move to board dts

NeilChen93 replied:
Thanks reminder, Updated.

NeilChen93 replied:
Hi @kartben , after move this to board dts, CI compilation report error, samples/sensor/thermometer build fail. This example also use aliases ambient-temp0, but frdm_mcxa156 don't support it. So cancel this update.

MaureenHelm replied:
I don't quite understand. It sounds like that might have been a legitimate build failure. Can you post a link to the CI run or paste the console output from a failed build?

ubieda replied:
Looked over this PR and I agree this should be moved over. The rest of the patches look good to me.

MaureenHelm reviewed (commented):
Code changes look ok, but need to address the CI failures.

NeilChen93 committed:
boards: nxp: frdm_mcxa156: Support i3c for NXP frdm_mcxa156 board

Support i3c for NXP frdm_mcxa156 board.
Test it using samples/sensor/thermometer.

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>

NeilChen93 committed:
boards: nxp: frdm_mcxa156: Add config TEST_RANDOM_GENERATOR

Add config TEST_RANDOM_GENERATOR to support function sys_rand_get

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>

henrikbrixandersen commented on boards/nxp/frdm_mcxa156/frdm_mcxa156_defconfig:
```diff
@@ -10,3 +10,4 @@ CONFIG_SERIAL=y
 CONFIG_UART_INTERRUPT_DRIVEN=y
 CONFIG_GPIO=y
 CONFIG_LPADC_DO_OFFSET_CALIBRATION=y
+CONFIG_TEST_RANDOM_GENERATOR=y
```
This should not be enabled for real hardware by default. How come this was added?
